src/Entity/BaseEntityManager.php

```diff
@@ -78,6 +80,25 @@ public function getEntityManager()
         return $objectManager;
     }
 
+    public function beginTransaction(): void
+    {
+        $this->getEntityManager()->beginTransaction();
+    }
+
+    public function commit(): void
+    {
+        try {
+            $this->getEntityManager()->commit();
+        } catch (\Throwable $exception) {
+            throw new TransactionException($exception->getMessage(), (int) $exception->getCode(), $exception);
```

Comment:
Have mixed feelings about wrapping in our own exception. What are the benefits? 

I have one con, people used to try catch normal doctrine exception for commit, will have to think about this new exception.

Reply:
To solve this we could extends from doctrine exception!
But I would say that usually will be used Or `Throwable` or `DoctrineException` into the code, I guess it won't be used!

IMO we could remove this try catch and let Doctrine exception raise!

Reply:
People are using commit on the Sonata EntityManager, not the doctrine one. So it would be weird to have doctrine exception.

Also, when you are working with the interface, ho do you know if it will be orm exception, mongo exception, something else exception etc.

The over option is to just use a throwable annotation... I thought it could be good to prefer a domain exception

Reply:
> Also, when you are working with the interface, ho do you know if it will be orm exception, mongo exception, something else exception etc.

In this case no, because we are implementing only for EntityManager!

> People are using commit on the Sonata EntityManager, not the doctrine one. So it would be weird to have doctrine exception.

Yeah I got your point, I agree that it's a bit strange implements a Sonata interface and throw an exception of other dependency!


well we have pros and cons 😄 

But if we going to use a custom Exception I guess we should extends of DoctrineException 👀 


Note: But looking for the benefit side, I would say it won't be too helpful, usually the catch will have only `DoctrineException` to catch some errors!

and maybe it will be a BC because for now we are throwing DoctrineException and if we return CustomException it won't go into the caches that already exist catching `Doctrine` stuffs.

Reply:
> In this case I'll know, because we are implementing only for EntityManager, mongo entity doesn't use this interface!

When I use this interface, I'm not supposed to know which one are implementing it and which one don't.
Maybe ORM will stop transaction one day. Maybe Mongo will start one day. 
The goal is too decouple from the implementation.

> But if we going to use a custom Exception I guess we should extends of DoctrineException 👀

I'm not sure you have something like a BaseDoctrineException, used by both ORM and Mongo.

> and maybe it will be a BC because for now we are throwing DoctrineException and if we return CustomException it won't go into the caches that already exist catching Doctrine stuffs.

It's not a BC break since we're adding a method/interface.
If you use `->getEntityManager->commit` you have the ORM exception
If you use `->commit` you have the Sonata exception